### PR TITLE
some cleanups and make sure strace filter for macos binary is working

### PR DESCRIPTION
**Pull request by ucgJhe**
as title.

### CODE REVIEW

File: qiling/os/macos/macos.py [hunk 1]
```diff
@@ -123,10 +123,10 @@ def runner(self):
                 self.ql.uc.emu_start(self.ql.entry_point, self.ql.until_addr, self.ql.timeout)
         except UcError:
             if self.ql.output in (QL_OUT_DEBUG, QL_OUT_DUMP):
-                self.ql.nprint("[+] PC= " + hex(self.ql.pc))
```
Comment (xwings) [inline qiling/os/macos/macos.py:126]: this is ok for nprint, already crash. So makes sense if we use nprint

File: qiling/os/macos/syscall.py [hunk 2]
```diff
@@ -42,15 +42,16 @@
 
 
 def ql_arm64_fgetattrlist(ql, fd, attrlist, attrbuff, attrsizebuff, options, *args, **kw):
-    ql.nprint("addr %x, path: %s" % (attrlist ,attrbuff))
+    ql.nprint("fgetattrlist(fd: %d, attrlist: 0x%x, attrbuff: 0x%x, attrsizebuff: 0x%x, options: 0x%x)" % (fd, attrlist, attrbuf, attrsizebuff, options))
+    ql.dprint(0, "addr: %x, path: %s" % (attrlist ,attrbuff))
     KERN_SUCCESS = 1
     ql_definesyscall_return(ql, KERN_SUCCESS)
-    ql.nprint("syscall[fgetattrlist] >> fgetattrlist")
+    ql.dprint(0, "syscall[fgetattrlist] >> fgetattrlist")
```
Comment (xwings) [inline qiling/os/macos/syscall.py:49]: just delete this. its a junk to me

File: qiling/os/macos/syscall.py [hunk 3]
```diff
@@ -42,15 +42,16 @@
...
 
 
 def ql_arm64_poll(ql, target, address, size, *args, **kw):
     ql_definesyscall_return(ql, KERN_SUCCESS)
-    ql.nprint("syscall[poll] >> poll")
+    ql.dprint(0, "syscall[poll] >> poll")
```
Comment (xwings) [inline qiling/os/macos/syscall.py:54]: just delete this. its a junk to me

File: qiling/os/macos/syscall.py [hunk 4]
```diff
@@ -338,14 +339,14 @@ def ql_syscall_mmap2_macos(ql, mmap2_addr, mmap2_length, mmap2_prot, mmap2_flags
 
 # 0xca
 def ql_syscall_sysctl(ql, name, namelen, old, oldlenp, new_arg, newlen):
-    ql.nprint("sysctl(name:0x{:X}, namelen:0x{:X}, old:0x{:X}, oldlenp:0x{:X}, new:0x{:X}, newlen:0x{:X})".format(
+    ql.nprint("sysctl(name: 0x{:X}, namelen: 0x{:X}, old: 0x{:X}, oldlenp: 0x{:X}, new: 0x{:X}, newlen: 0x{:X})".format(
```
Comment (xwings) [inline qiling/os/macos/syscall.py:342]: all these we should change to format string output, 3.5 compatible printout.

Comment (ucgJhe) [reply qiling/os/macos/syscall.py:342]: sure

Comment (xwings) [reply qiling/os/macos/syscall.py:342]: lots of them tho, this can be next PR. if u agree

File: qiling/os/macos/syscall.py [hunk 5]
```diff
@@ -338,28 +337,28 @@ def ql_syscall_mmap2_macos(ql, mmap2_addr, mmap2_length, mmap2_prot, mmap2_flags
...
+    ql.nprint("sysctlbyname(name: 0x%X, namelen: 0x%X, old: 0x%X, oldlenp: 0x%X, new: 0x%X, newlen: 0x%X)" % (
         name, namelen, old, oldlenp, new_arg, newlen
     ))
     ql_definesyscall_return(ql, KERN_SUCCESS)
 
 # 0x126
 # check shared region if avalible , return not ready every time
 def ql_syscall_shared_region_check_np(ql, p, uap, retvalp, *args, **kw):
-    ql.nprint("shared_region_check_np(p: {}, uap: {}, retvalp :{}) : ret:{}".format(p, uap, retvalp, EINVAL))
+    ql.nprint("shared_region_check_np(p: 0x%X, uap: 0x%X, retvalp: 0x%X) : ret: 0x%X" % (p, uap, retvalp, EINVAL))
```
Comment (xwings) [inline qiling/os/macos/syscall.py:355]: Question: should it be 0x%X or 0x%X ?

Comment (ucgJhe) [reply qiling/os/macos/syscall.py:355]: 0x%X for uppercase hex and 0x%x for lowercase hex, depends which you prefer

Comment (xwings) [reply qiling/os/macos/syscall.py:355]: since we are all using lower case. make it a standard.

Comment (ucgJhe) [reply qiling/os/macos/syscall.py:355]: how about the parameter names ?
should we print it out or just plain value like shared_region_check_np(X , X , X)

Comment (xwings) [reply qiling/os/macos/syscall.py:355]: i think take out the param too, make it standaized. the param can leave it dprint, just like how u did it previously for those file open

**Comment by xwings** [comment]
thanks !

**Pull request merged**